pkg/controller/ads/extensions/local_ratelimit.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extensions
+
+import (
+	"fmt"
+	"time"
+
+	v1 "github.com/cncf/xds/go/udpa/type/v1"
+	listenerv3 "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"kmesh.net/kmesh/api/v2/filter"
+	"kmesh.net/kmesh/api/v2/listener"
+)
+
+const LocalRateLimit = "envoy.filters.tcp.local_ratelimit"
+
+// NewLocalRateLimit constructs a new LocalRateLimit filter wrapper.
+func NewLocalRateLimit(filter *listenerv3.Filter) (*listener.Filter_LocalRateLimit, error) {
+	localRateLimit, err := newLocalRateLimit(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &listener.Filter_LocalRateLimit{
+		LocalRateLimit: localRateLimit,
+	}, nil
+}
+
+// newLocalRateLimit creates a new LocalRateLimit filter.
+func newLocalRateLimit(Filter *listenerv3.Filter) (*filter.LocalRateLimit, error) {
+	unstructured, err := unmarshalToTypedStruct(Filter)
```

Comment:
why donot you import xds and use structured api

Reply:
FYI https://github.com/istio/istio/blob/master/pkg/wasm/convert.go#L145-L167

TypedExtensionConfig can also be unmarshaled

Reply:
``` go
func main() {
	bucket := &v1.TypedStruct{
		TypeUrl: "type.googleapis.com/envoy.extensions.filters.http.local_ratelimit.v3.LocalRateLimit",
		Value: &structpb.Struct{
			Fields: .....
		},
	}

	anyTyped, _ := anypb.New(bucket)
	extConfig := &corev3.TypedExtensionConfig{}
	if err := anyTyped.UnmarshalTo(extConfig); err != nil { // panic
		panic(err)
	}

	typedStruct := &v1.TypedStruct{}
	if err := anyTyped.UnmarshalTo(typedStruct); err != nil { // no panic
		panic(err)
	}
}
```

I tested it, and it seems it cannot unmarshal into TypedExtensionConfig.

Panic: proto: mismatched message type: got "envoy.config.core.v3.TypedExtensionConfig", want "udpa.type.v1.TypedStruct"

Istio is using TypedStruct, I don't understand how to modify it.